Entity/CampaignClientStatsRepository.php

```diff
@@ -212,21 +212,21 @@ public function getExistingEntitiesByDate($params)
      */
     public function updateExistingEntitiesByDate($params, EntityManager $em)
     {
-        $conn           = $em->getConnection();
-        $origAutoCommit = $conn->isAutoCommit();
-        $conn->setAutoCommit(true);
-
-        $qb = $conn->createQueryBuilder();
-        $qb->update($this->getTableName(), 's')
-            ->set('s.reprocess_flag', true)
-            ->where(
-                $qb->expr()->eq('s.reprocess_flag', false),
-                's.date_added = FROM_UNIXTIME(:dateAdded)'
-            )
-            ->setParameter('dateAdded', $params['dateTo'])
-            ->execute();
-
-        $conn->setAutoCommit($origAutoCommit);
+        $now      = new \DateTime();
+        $safeTime = $now->sub(new \DateInterval('PT15M'))->getTimestamp();
+
+        if ($params['dateTo'] < $safeTime) {
+            $qb = $em->getConnection()->createQueryBuilder();
+            $qb->update($this->getTableName(), 's')
+                ->set('s.reprocess_flag', true)
+                ->where(
+                    $qb->expr()->eq('s.reprocess_flag', 0),
+                    $qb->expr()->eq('s.date_added', 'FROM_UNIXTIME(:dateAdded)')
+                )
+                ->setParameter('dateAdded', $params['dateTo'])
+                ->setMaxResults(1)
```

Comment:
setMaxResults() only works on select queries
https://stackoverflow.com/questions/44216276/is-there-a-way-to-add-a-limit-to-an-update-query-in-doctrine-orm?rq=1

You'll have to have a WHERE to limit it, maybe something like 
`WHERE s.id IN (SELECT id FROM reprocess_flag WHERE date_added = :dateAdded LIMIT 1)`

Reply:
Update: I'm an idiot, LIMIT is fine on UPDATE queries.

Reply:
There is a lot of conflicting info on the interwebs, but it appears that Doctrine 2 ORM allows it. But maybe not strict DQL (or mysql).